tests/Metalama.Framework.Tests.UnitTests/DesignTime/Pipeline/DesignTimePipelineTests.cs

```diff
@@ -2314,4 +2314,83 @@ public override void BuildAspect(IAspectBuilder<ICompilation> builder)
 
         Assert.Empty( result.GetAllDiagnostics() );
     }
+
+    [Fact]
+    public void IncompleteGenericArgumentInTransformationTarget()
+    {
+        const string aspect =
+            """
+            using Metalama.Framework.Advising;
+            using Metalama.Framework.Aspects;
+            using Metalama.Framework.Code;
+
+            public class TestAspect : ConstructorAspect
+            {
+                public override void BuildAspect(IAspectBuilder<IConstructor> builder)
+                {
+                    builder.IntroduceParameter("TestParameter", typeof(int), TypedConstant.Create(1));
+                }
+            }
+            """;
+
+        const string target1 =
+            """
+            using System.Collections.Generic;
+
+            public partial class TargetCode
+            {
+                [TestAspect]
+                public TargetCode(List<int> x)
+                {
+                }
+            }
+            """;
+
+        const string target2 =
+            """
+            using System.Collections.Generic;
+
+            public partial class TargetCode
+            {
+                [TestAspect]
+                public TargetCode(List<List<>> x)
+                {
+                }
+            }
+            """;
+
+        const string target3 =
+            """
+            using System.Collections.Generic;
+
+            public partial class TargetCode
+            {
+                [TestAspect]
+                public TargetCode(List<List<int>> x)
+                {
+                }
+            }
+            """;
+
+        using var testContext = this.CreateTestContext();
+
+        var code1 = new Dictionary<string, string> { ["aspect.cs"] = aspect, ["target.cs"] = target1 };
+        var code2 = new Dictionary<string, string> { ["aspect.cs"] = aspect, ["target.cs"] = target2 };
+        var code3 = new Dictionary<string, string> { ["aspect.cs"] = aspect, ["target.cs"] = target3 };
+
+        var compilation1 = CreateCSharpCompilation( code1, assemblyName: "test", acceptErrors: true );
+        var compilation2 = CreateCSharpCompilation( code2, assemblyName: "test", acceptErrors: true );
+        var compilation3 = CreateCSharpCompilation( code3, assemblyName: "test", acceptErrors: true );
+
+        using TestDesignTimeAspectPipelineFactory factory = new( testContext );
+
+        //Assert.True( factory.TryExecute( testContext.ProjectOptions, compilation1, default, out var result1 ) );
+        //Assert.Single( result1.Result.IntroducedSyntaxTrees);
```

Comment:
This bug only occurs when you have a valid state and then get the pipeline into an invalid state, during `AspectPipelineResult.Update`. The third case is there just for completeness, so that the pipeline manages to recover from the error (no particular reason). The crash originally occurred during the second `TryExecute`. I'll add comments